modules/Core/src/main/java/org/terasology/core/world/generator/rasterizers/SolidRasterizer.java

```diff
@@ -93,51 +94,53 @@ public void generateChunk(CoreChunk chunk, Region chunkRegion) {
-                    return grass;
-                } else if (depth == 0 && height >= seaLevel + 96) {
-                    return snow;
-                } else if (depth > 32) {
-                    return stone;
-                } else {
-                    return dirt;
-                }
-            case SNOW:
-                if (depth == 0 && height > seaLevel) {
-                    // Snow on top
-                    return snow;
-                } else if (depth > 32) {
-                    // Stone
-                    return stone;
-                } else {
-                    // Dirt
-                    return dirt;
-                }
-            case DESERT:
-                if (depth > 8) {
-                    // Stone
-                    return stone;
-                } else {
-                    return sand;
-                }
-            case OCEAN:
-                if (depth == 0) {
-                    return sand;
-                } else {
-                    return stone;
-                }
-            case BEACH:
-                if (depth < 3) {
-                    return sand;
-                } else {
-                    return stone;
-                }
+    private Block getSurfaceBlock(int depth, int height, Biome type, int seaLevel) {
+        if (type instanceof CoreBiome) {
+            switch ((CoreBiome) type) {
+                case FOREST:
+                case PLAINS:
+                case MOUNTAINS:
+                    // Beach
+                    if (depth == 0 && height > seaLevel && height < seaLevel + 96) {
+                        return grass;
+                    } else if (depth == 0 && height >= seaLevel + 96) {
+                        return snow;
+                    } else if (depth > 32) {
+                        return stone;
+                    } else {
+                        return dirt;
+                    }
+                case SNOW:
+                    if (depth == 0 && height > seaLevel) {
+                        // Snow on top
+                        return snow;
+                    } else if (depth > 32) {
+                        // Stone
+                        return stone;
+                    } else {
+                        // Dirt
+                        return dirt;
+                    }
+                case DESERT:
+                    if (depth > 8) {
+                        // Stone
+                        return stone;
+                    } else {
+                        return sand;
+                    }
+                case OCEAN:
+                    if (depth == 0) {
+                        return sand;
+                    } else {
+                        return stone;
+                    }
+                case BEACH:
+                    if (depth < 3) {
+                        return sand;
+                    } else {
+                        return stone;
+                    }
+            }
         }
         return dirt;
```

Comment:
Maybe it should not return dirt if `type` is not a `CoreBiome`. I don't like returning `null` either, but at least it would be obvious that there is no mapping. What do you think?